core/domain/summary_services.py

```diff
@@ -256,3 +256,18 @@ def get_top_rated_exploration_summary_dicts(language_codes):
         reverse=True)[:NUMBER_OF_TOP_RATED_EXPLORATIONS]
 
     return _get_displayable_exp_summary_dicts(sorted_exp_summaries)
+
+
+def get_recently_published_exploration_summary_dicts(language_codes):
+    """Returns a list of recently published explorations with the given language code.
+    """
+    collect_all_exp_summaries = [
```

Comment:
This implementation is problematic, actually, because it is really slow. You're getting all the summaries, and you are sorting all of them -- all in order to get a final list of only 8 summaries.

Instead, can you take a look at get_featured_exploration_summaries() and do something similar to what's done there? Note that (if you follow the chain down into the storage layer) this ends up making use of a datastore query that retrieves the exploration summaries with featured status, and only those exploration summaries, from the datastore. You'll want to do something similar.